src/compile/scale/parse.ts

```diff
@@ -47,25 +48,37 @@ export function parseScale(model: UnitModel, channel: ScaleChannel) {
   const scale = model.scale(channel);
   const sort = model.sort(channel);
 
-  const scaleComponent: VgScale = {
-    name: model.scaleName(channel + '', true),
-    type: scale.type,
-    domain: parseDomain(model, channel),
-    range: parseRange(scale)
-  };
+  // FIXME parseScale should return something similar to Split<VgScale>
+  const scaleComponent: ScaleComponent = new Split<Partial<VgScale>>(
+    {},
+    // Implicit
+    {
+      name: model.scaleName(channel + '', true)
+    }
+  );
+  scaleComponent.copyKeyFrom('type', scale);
+
+  // FIXME: this condition to be implicit/explicit is incorrect
```

Comment:
Because it requires refactoring parseDomain, which would become hairy ... I wanna work on more important parts first.  Filed an issue here https://github.com/vega/vega-lite/issues/2497